### PR TITLE
feat: backend migration to connect-rpc

### PR DESCRIPTION
**Pull request by strantalis**
### Proposed Changes

Implementation of the following adr https://github.com/opentdf/platform/issues/1524

The breaking changes are from changing the Idempontency Level. This is required if we would like to support GET request on some of our services. 

https://connectrpc.com/docs/go/get-requests-and-caching

### Checklist

- [x] I have added or updated unit tests
- [x] I have added or updated integration tests (if appropriate)
- [ ] I have added or updated documentation

### CODE REVIEW

**Review by pflynn-virtru — DISMISSED** [review]

File: service/internal/server/server.go [hunk 1]
```diff
@@ -195,25 +197,51 @@ func NewOpenTDFServer(config Config, logger *logger.Logger) (*OpenTDFServer, err
...
+	code        int
+	wroteHeader bool
+}
+
+func (rw *grpcGatewayResponseWriter) Header() http.Header {
+	return rw.w.Header()
+}
+
+func (rw *grpcGatewayResponseWriter) WriteHeader(statusCode int) {
+	if rw.wroteHeader == false {
```
Comment (jrschumacher) [inline service/internal/server/server.go:212]:
```suggestion
	if !rw.wroteHeader {
```

File: service/internal/server/server.go [hunk 2]
```diff
@@ -104,13 +101,19 @@ type CORSConfig struct {
 	MaxAge           int      `mapstructure:"maxage" json:"maxage" default:"3600"`
 }
```
Comment (jrschumacher) [inline service/internal/server/server.go:103]:
```suggestion
const GRPCGatewayDeprecationDate = fmt.Sprintf("@%d", time.Date(2025, time.March, 25, 0, 0, 0, 0, time.UTC).Unix())

```

File: service/internal/server/server.go [hunk 3]
```diff
@@ -195,25 +197,51 @@ func NewOpenTDFServer(config Config, logger *logger.Logger) (*OpenTDFServer, err
...
+	wroteHeader bool
+}
+
+func (rw *grpcGatewayResponseWriter) Header() http.Header {
+	return rw.w.Header()
+}
+
+func (rw *grpcGatewayResponseWriter) WriteHeader(statusCode int) {
+	if rw.wroteHeader == false {
+		rw.w.Header().Set("Deprecation", fmt.Sprintf("@%d", time.Date(2025, time.March, 25, 0, 0, 0, 0, time.UTC).Unix()))
```
Comment (jrschumacher) [inline service/internal/server/server.go:213]:
```suggestion
		rw.w.Header().Set("Deprecation", GRPCGatewayDeprecationDate)
```

File: service/internal/server/server.go [hunk 4]
```diff
@@ -230,37 +258,60 @@ func newHTTPServer(c Config, h http.Handler, a *auth.Authentication, g *grpc.Ser
...
+		// Apply CORS to connectRPC and extra handlers
+		connectRPC = corsHandler.Handler(connectRPC)
+		grpcGateway = corsHandler.Handler(grpcGateway)
+	}
+
+	// Add authN interceptor to extra handlers
+	if c.Auth.Enabled {
+		grpcGateway = a.MuxHandler(grpcGateway)
+	} else {
+		l.Error("disabling authentication. this is deprecated and will be removed. if you are using an IdP without DPoP set `enforceDPoP = false`")
```
Comment (jrschumacher) [inline service/internal/server/server.go:272]: We should consider using hmac for a minimally safe no-oidc system or just push deprecation of this.

File: service/internal/server/server.go (hunk 2 above)
Comment (strantalis) [reply service/internal/server/server.go:103]: You can't do a constant like this in go.

File: service/internal/server/server.go (hunk 4 above)
Comment (strantalis) [reply service/internal/server/server.go:272]: It is currently nice for development but a simple hmac token setup would also make development easy.

**Review by pflynn-virtru — APPROVED** [review]

**Pull request merged**